### ISSUE TITLE
[Bug] Multibinding in a data template doesn't bind to current ItemSource 

### ISSUE THREAD

**Issue by wbsimms** [xamarin/Xamarin.Forms#11135]
### Description
I have a Collection view with an ItemsSource. Using multibinding in the DataTemplate fails when asking to bind to the item. 

### Steps to Reproduce

```
public class ViewModel { 
 public ObservableCollection<SampleData> SomeObservableCollection {get ;set}}
 public string SomeProperty {get;set;}
}

public class SampleData {
 public string ThisWontBind {get;set;}
}

<CollectionView Grid.Row="1" x:Name="ArtistsListView" ItemsSource="{Binding SomeObservableCollection}">
...
                    <CollectionView.ItemTemplate>
                        <DataTemplate>
                            <Grid>
                                <Grid.RowDefinitions>
                                    <RowDefinition Height="Auto"></RowDefinition>
                                    <RowDefinition Height="Auto"></RowDefinition>
                                </Grid.RowDefinitions>
                                <ImageButton Grid.Row="0">
                                    <ImageButton.Source>
                                        <MultiBinding Converter="{converters:StringJoinerConverter}">
                                            <Binding Path="SomeProperty" Source="{StaticResource Locator}" /> <!-- Binding here works -->
                                            <Binding Path="ThisWontBind" /> <!-- Binding here fails -->
                                        </MultiBinding>
                                    </ImageButton.Source>
                                </ImageButton>
TextColor="White"></Label>
                            </Grid>
                        </DataTemplate>
                    </CollectionView.ItemTemplate>
                </CollectionView>
```

### Expected Behavior
I would expect the binding the respect the current item in the ItemsSource

### Actual Behavior
Binding fails w/ null

### Basic Information

- Version with issue:  4.7.0
- Last known good version: New feature
- IDE: VS2019
- Platform Target Frameworks: All

### Screenshots
N/A

### Reproduction Link
See above

### Workaround
None

**Comment by StephaneDelcroix** [comment]
Could you please attach a real, contained and complete project for this ? We're missing too many information here to triage this. Thanks

**Comment by wbsimms** [comment]
Repo here:
https://github.com/DevFoundries/XamarinMultibinding11135

StringFormat is working. However, when using a IMultiConverter, it's not working in my example.

**Comment by StephaneDelcroix** [comment]
I downloaded your repro, and the multibinding works fine. Getting 'null' values is expected at first, when the BindingContext is 'null'.

The code throws a bit further away because your converter returns a `string`, and that `string` is set as the value of the `ImageSource`.

This isn't really a bug, but an inconsistency, and a serious pain in the back, so it has been already fixed in #11065 and will be part of the 4.7.0-sr1 coming out very very soon.

I'm closing this report as I believe it's already been fixed. Make sure to try again with the service release, and report anything going wrong with it.
